clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -742,25 +774,29 @@ protected interface ResponseHandler<T> {
     }
 
     /**
-     * Manages completing the future for a synchronous acknowledgement commit by counting
+     * Sends a ShareAcknowledgeCommitCallback event to the application when it is done
+     * processing all the remaining acknowledgement request states.
+     * Also manages completing the future for synchronous acknowledgement commit by counting
      * down the result as they are known and completing the future at the end.
      */
-    static class CommitSyncResultHandler {
+    class CommitResultHandler {
```

Comment:
I don't think this class needs to be related to an instance of the containing class. As a result, I would make it a static inner class.

Reply:
The class uses a non-static field(backgroundEventHandler) inside to add the background event. So I made the class non-static.